src/workloads/query/QueryStats.yml

```diff
@@ -10,233 +10,164 @@ KeyWords:
-        Collection: *Collection
-        Operations:
-        - OperationName: find
-          OperationCommand:
-            # This will create a find command with a random number of fields from 3 to 350, to create varying query shapes and sizes.
-            Filter:
-              {$and:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-- Name: Quiesce
-  Type: QuiesceActor
-  Threads: 1
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [5]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Repeat: 1
-
-- Name: FindConcurrentlyWithQueryStatsWithHMAC
-  Type: CrudActor
-  Threads: 32
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [6]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Duration: 1 minute
-        Collection: *Collection
-        Operations:
-        - OperationName: find
-          OperationCommand:
-            Filter:
-              {$and:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-# QueryStats with HMAC.
-- ActorFromTemplate:
-    TemplateName: QueryStatsAggregation
-    TemplateParameters:
-      Name: "QueryStatsAggregationWithHMAC"
-      queryStatsParameters: {transformIdentifiers: {algorithm: "hmac-sha-256", hmacKey: *binDataGen}}
-      OnlyActiveInPhase: 6
-
-- Name: Quiesce
-  Type: QuiesceActor
-  Threads: 1
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [7]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Repeat: 1
-
-- Name: FindConcurrentlyWithQueryStatsWithoutHMAC
-  Type: CrudActor
-  Threads: 32
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [8]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Duration: 1 minute
-        Collection: *Collection
-        Operations:
-        - OperationName: find
-          OperationCommand:
-            Filter:
-              {$and:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-# QueryStats without HMAC.
-- ActorFromTemplate:
-    TemplateName: QueryStatsAggregation
-    TemplateParameters:
-      Name: "QueryStatsAggregationWithoutHMAC"
-      queryStatsParameters: {}
-      OnlyActiveInPhase: 8
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
```

Comment:
Most people will come to this point in the workload file to decipher the flow of the workload phases, and even though you've written them out in a useful order, it doesn't make it clear in what phases `ClearCollection`, `InsertData`, or some of the actors run. My suggestion would be to make "Phases" a parameter just like it is for the other actors. Even though the parameter would be technically unnecessary it will make it easier to follow. Alternatively, you could add a comment here to clarify.